tardis/apps/publication_workflow/components/Forms/SelectDatasetForm.jsx

```diff
@@ -0,0 +1,215 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import {
+  Card,
+  Col, Form, Row, Table,
+} from 'react-bootstrap';
+
+import {
+  fetchDatasetsForExperiment,
+  fetchExperimentList,
+} from '../../../../../assets/js/apps/tiles/components/utils/FetchData';
+
+const SelectDatasetForm = ({ formik }) => {
+  const [expList, setExpList] = useState([]);
+  const [datasetList, setDatasetList] = useState([]);
+  const [selectedDatasetList, setSelectedDatasetList] = useState(formik.values.selectedDatasets);
+  const [selectedExperiment, setSelectedExperiment] = useState();
+
+  const handleExpChange = (event) => {
+    event.preventDefault();
+    fetchDatasetsForExperiment(event.target.value).then(result => setDatasetList(result));
+    setSelectedExperiment(expList[event.target.value]);
+  };
+  const handleDatasetSelectChange = (event) => {
+    const selectedOptionsArray = [].slice.call(event.target.selectedOptions);
+    let currentlySelectedDatasetList = [];
+    selectedOptionsArray
+      .map(item => (currentlySelectedDatasetList
+        .push({
+          experiment: selectedExperiment.title,
+          experiment_id: selectedExperiment.id,
+          dataset: {
+            id: item.value,
+            description: item.label.slice(0, item.label.length),
+          },
+        })));
+    currentlySelectedDatasetList = selectedDatasetList.concat(currentlySelectedDatasetList);
+    setSelectedDatasetList(currentlySelectedDatasetList);
+    // run field validation
+    formik.validateField('selectedDatasets');
+  };
+
+  useEffect(() => {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 2.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933947)
```suggestion
    useEffect(() => {
```